server/src/main/java/org/apache/druid/metadata/SQLMetadataSupervisorManager.java

```diff
@@ -249,6 +252,34 @@ public Void withHandle(Handle handle) throws Exception
     );
   }
 
+  @Override
+  public int removeTerminatedSupervisorsOlderThan(long timestamp)
+  {
+    int removedCount = 0;
+    DateTime dateTime = DateTimes.utc(timestamp);
+    Map<String, SupervisorSpec> supervisors = getLatest();
+    for (Map.Entry<String, SupervisorSpec> supervisor : supervisors.entrySet()) {
+      final SupervisorSpec spec = supervisor.getValue();
+      if (spec instanceof NoopSupervisorSpec) {
+        removedCount += dbi.withHandle(
+            handle -> {
+              Update sql = handle.createStatement(
+                  StringUtils.format(
+                      "DELETE FROM %1$s WHERE spec_id = :spec_id AND created_date < :date_time",
```

Comment:
does it make sense to delete in batch than doing one by one which will be slow?

Reply:
Done